sda-commons-server-key-mgmt/README.md

```diff
@@ -0,0 +1,64 @@
+# sda-commons-server-key-mgmt
+Bundle for key management in microservices.
+The bundle provide means to work with keys in APIs and to retrieve mapping values from or to a possible implementation depending on the keys.
+This includes
+  * checking of a key is valid
+  * checking if a value is valid for a given key
+  * mapping of a key between API and implementation specific values
+
+API keys and values should be defined in snake case. For these two values, the bundle will be case tolerant, by mapping to uppercase internally.
+For implementations specific values, the bundle must not be case tolerant to provide keys as expected.
+
+## Usage
+```
+    implementation 'org.sdase.commons:sda-commons-server-key-mgmt:<version>'
+```
+
+Initialization of bundle:
+```java    
+  private final KeyMgmtBundle<KeyMgmtBundleTestConfig> keyMgmt =
+        KeyMgmtBundle.builder()
+            .withKeyMgmtConfigProvider(KeyMgmtBundleTestConfig::getKeyMgmt)
+            .build();
+```
+
+The [configuration](src/main/java/org/sdase/commons/keymgmt/config/KeyMgmtConfig.java) includes the paths to the mapping and keys yaml files.
+
+### Key yaml file
+The yaml file or keys may contain one or more [KeyDefinition](src/main/java/org/sdase/commons/keymgmt/model/KeyDefinition.java) documents.
+
+__Example:__
+```yaml
+name: GENDER
```

Comment:
Using `name` here and `key` in the signature of the mapping methods confuses me. What do you think about `type` for this?

Reply:
I dont like `type` since it is soooooo generic. I understand your point. Why not using `key` in both places?

Reply:
Although you do not use key for the mapping of api and impl here explicitly, in my mind the actual `ValueMappings` related to a `KeyDefinition` are key value pairs like in a Java `Map`. And then from my perspective the term `key` is gone.

Anyways, naming both locations on the same bases leads to the right direction. What do you think about to keep `name` in the definition (like the name of a Java Type) and use `keyDefinitionName` in the method signatures to clarify where the value is specified?

Reply:
using keyDefinitionName now